plugin/src/main/java/org/astropeci/omw/item/PeriodicItemDispenser.java

```diff
@@ -113,6 +106,30 @@ private void giveItemsPeriodically() {
         Bukkit.getScheduler().scheduleSyncDelayedTask(plugin, this::giveItemsPeriodically, ITEM_DROP_DELAY);
     }
 
+    @SneakyThrows({InvalidConfigurationException.class})
+    private Set<ItemStack> getItemsFromConfig()
+    {
+        Set<ItemStack> set = new HashSet<ItemStack>();
+        List<Map<?, ?>> items = plugin.getConfig().getMapList("items");
+        for(Map<?, ?> i : items){
+            var m = (Map<String, ?>) i;
+            ItemStack item;
+            switch ((String)m.get("type")){
+                case "simpleItem":
+                    item = simpleItem(Material.getMaterial((String) m.get("material")), (String)m.get("name"));
+                    break;
+                case "itemStack":
+                    item = new ItemStack(Objects.requireNonNull(Material.getMaterial((String) m.get("material"))), (int)m.get("amount"));
+                    break;
+                default:
+                    throw new InvalidConfigurationException("Item type has to be either 'simpleItem' or 'itemStack'");
+            }
+
+            set.add(item);
+        }
+        return set;
+    }
```

Comment:
I'm also not sure the `simpleItem`/`itemStack` system is a good idea for public configuration, I'd prefer if it were a more general system. I'd also prefer if usage of `var` was avoided.`set` could also probably be renamed to something slightly more descriptive such as `itemSet` or `result`.

Reply:
This should probably use a system where every item has to have a material and all other properties are optional.
Also, as more of a Haskell developer, I'd be interested to hear why you would want to avoid `var`. I'll definitely try to avoid it either way.

Reply:
No good reason really, its more just that the rest of the project doesn't use `var` so might seem out of place to adopt in small sections of the code.